include/eve/module/core/regular/impl/convert.hpp

```diff
@@ -139,4 +126,179 @@ namespace eve::detail
       }
     }
   }
+
+  //================================================================================================
+  // wide<->wide convert
+
+  //================================================================================================
+  // wide<->wide default convert implementation
+  template<typename In, typename Out>
+  EVE_FORCEINLINE auto convert_impl(EVE_SUPPORTS(cpu_), In const &v0, as<Out> const &tgt) noexcept
+  {
+    using out_t = as_wide_t<Out, cardinal_t<In>>;
+
+    if constexpr(has_aggregated_abi_v<In>)
+    {
+      // If input is aggregated, we can slice and combine without lose of performance
+      return out_t{eve::convert(v0.slice(lower_),tgt),eve::convert(v0.slice(upper_),tgt)};
+    }
+    else
+    {
+      return map(convert,v0,tgt);
+    }
+  }
+
+  template<simd_value In, scalar_value Out>
+  requires( !product_type<In> && !product_type<Out>)
+  EVE_FORCEINLINE auto convert_(EVE_SUPPORTS(cpu_), In const &v0, as<Out> const &tgt) noexcept
```

Comment:
This convert is in `impl`. 
I don't think the assertions you wrote will actually trigger if you have a specialised conversion.

Reply:
Assertion removed